subsys/stats/Kconfig

```diff
@@ -0,0 +1,27 @@
+# Kconfig - mcumgr configuration options
```

Comment:
"Statistics can be retrieved with the mcumgr management subsystem."

I'm not entirely sure how this stats subsystem works, but if it is only useful for mcumgr-enabled applications then perhaps it should be located within subsys/mcumgr?

At least it should be reflected in the Kconfig how mcumgr and stats depend on each other.
Perhaps there should be an

`if MCUMGR 
`

here.

Reply:
Right now it is at the toplevel of the Kconfig tree, at the same level as Networking, Device Drivers, File Systems, etc.

Reply:
Putting statistics at the top level of the Kconfig tree does feel a bit wrong.  I don't think it belongs with mcumgr, because statistics should be accessible by means other than mcumgr (e.g., shell).  I mentioned mcumgr only in the Kconfig help because that is the only retrieval mechanism that is currently implemented.

How about if I move the statistics Kconfig to the "System Monitoring Options" menu?

Re: placement in the source tree- I hate to create a new directory under `subsys` for this, but I don't see a good home for the stats code.  I could move it to `misc` where the system monitoring Kconfig settings are defined, but that feels more chaotic.  Do you have any suggestions?

Reply:
"How about if I move the statistics Kconfig to the "System Monitoring Options" menu?"
Sounds good.

"Do you have any suggestions?"
subsys is fine as far as I am concerned. @nashif might know better.